src/com/xilinx/rapidwright/rwroute/RWRoute.java

```diff
@@ -849,7 +848,8 @@ private void setRerouteCriticality() {
 			criticalConnections.sort(new Comparator<Connection>() {
 				@Override
 				public int compare(Connection connection1, Connection connection2) {
-					return connection1.getCriticality() < connection1.getCriticality()? 1 : -1;
+					if(connection1.getCriticality() == connection2.getCriticality()) return 0;
+					else return connection1.getCriticality() < connection2.getCriticality()? 1 : -1;
```

Comment:
```suggestion
					return Float.compare(connection1.getCriticality(),connection2.getCriticality());
```
@YunxZhou - would this code be simpler?  Is it important that the return values be `-1` and `1` or does it just need to be negative and positive integers?